ipfs-cluster-service/state.go

```diff
@@ -44,11 +42,62 @@ func upgrade() error {
 	}
 
 	// Reset with SnapshotSave
-	err = raft.SnapshotSave(consensusCfg, newState, clusterCfg.ID)
+	return raft.SnapshotSave(consensusCfg, newState, clusterCfg.ID)
+}
+
+func export(output string) error {
```

Comment:
most of this function is very similar to upgrade() isn't it?

should be abstracted into `restoreStateFromDisk` (which runs state.Restore() (or Migrate())?

and then upgrade() and export() can re-use that, one of them saving a new snapshot and the other writing to disk the state.